src/Compilers/CSharp/Test/Emit/Emit/EditAndContinue/EditAndContinueClosureTests.cs

```diff
@@ -125,6 +125,107 @@ void F()
                 Row(5, TableIndex.MethodDef, EditAndContinueOperation.Default));
         }
 
+        [Fact(Skip = "https://github.com/dotnet/roslyn/issues/43099")]
+        public void MethodWithSwitchExpression()
+        {
+            var source0 = MarkedSource(@"
+using System;
+
+class C
+{
+    int F(object o)
+    {
+        return o switch
+        {
+            int i => new Func<int>(<N:0>() => i + (int)o + i switch
+                {
+                    1 => 1,
+                    _ => new Func<int>(<N:1>() => (int)o + 1</N:1>)()
+                }</N:0>)(),
+            _ => 0
+        };
+    }
+}");
+            var source1 = MarkedSource(@"
+using System;
+
+class C
+{
+    int F(object o)
+    {
+        return o switch
+        {
+            int i => new Func<int>(<N:0>() => i + (int)o + i switch
+                {
+                    1 => 1,
+                    _ => new Func<int>(<N:1>() => (int)o + 2</N:1>)()
+                }</N:0>)(),
+            _ => 0
+        };
+    }
+}");
+            var compilation0 = CreateCompilation(source0.Tree, options: ComSafeDebugDll.WithMetadataImportOptions(MetadataImportOptions.All));
+            var compilation1 = compilation0.WithSource(source1.Tree);
+
+            var v0 = CompileAndVerify(compilation0);
+            var md0 = ModuleMetadata.CreateFromImage(v0.EmittedAssemblyData);
+
+            var f0 = compilation0.GetMember<MethodSymbol>("C.F");
+            var f1 = compilation1.GetMember<MethodSymbol>("C.F");
+
+            var generation0 = EmitBaseline.CreateInitialBaseline(md0, v0.CreateSymReader().GetEncMethodDebugInfo);
+
+            var diff1 = compilation1.EmitDifference(
+                generation0,
+                ImmutableArray.Create(SemanticEdit.Create(SemanticEditKind.Update, f0, f1, GetSyntaxMapFromMarkers(source0, source1), preserveLocalVariables: true)));
+
+            // TODO: Below is incorrect and needs updating once compiler bugs are fixed
```

Comment:
>TODO [](http://example.com/codeflow?start=15&length=4)

Is there an issue id that could be used here and below instead of TODOs?

Reply:
Oops, forgot about these. Thank you for noticing!

These need to be fixed in this PR, but didn't want to do so until the compiler bugs were fixed so we could be sure the closure info is correct. Putting this back to draft to work on it some more!